game-core/src/main/java/games/strategy/engine/data/UnitTypeList.java

```diff
@@ -40,6 +44,22 @@ public UnitType getUnitType(final String name) {
     return types;
   }
 
+  /**
+   * Returns the unit support rules for the unit types. Computed once and cached afterwards.
+   *
+   * @return The unit support rules.
+   */
+  public Set<UnitSupportAttachment> getSupportRules() {
+    if (supportRules == null) {
+      supportRules =
+          UnitSupportAttachment.get(getData())
+              .parallelStream()
+              .filter(usa -> (usa.getRoll() || usa.getStrength()))
```

Comment:
Are you sure that `usa`, unit support attachment `getStrength` will never change?

Reply:
I am not actually. If it can change, then there should be logic to invalidate the cached info.

Do you know who the right person would be to give a definitive answer?

Reply:
It looks like it will not change after a game data has been loaded:

UnitSupportAttachment.java:
```
        .put("strength", MutableProperty.ofReadOnly(this::getStrength))
```
```
  public static <T> MutableProperty<T> ofReadOnly(final Supplier<T> getter) {
    return of(noSetter(), noStringSetter(), getter, noResetter());
  }
```

```
    <attachment name="supportAttachmentFortificationbritishex" attachTo="britishFortification" javaClass="games.strategy.triplea.attachments.UnitSupportAttachment" type="unitType">
      <option name="unitType" value="britishInfantry:britishMarine:britishCombatEngineer:britishAlpineInfantry:britishParatrooper:russianInfantry:russianMarine:russianCombatEngineer:russianAlpineInfantry:russianParatrooper:americanInfantry:americanMarine:americanCombatEngineer:americanAlpineInfantry:americanParatrooper:chineseInfantry:chineseCombatEngineer:chineseAlpineInfantry"/>
      <option name="faction" value="allied"/>
      <option name="side" value="defence"/>
      <option name="dice" value="strength"/>
      <option name="bonus" value="2"/>
      <option name="number" value="3"/>
      <option name="bonusType" value="FortBonus"/>
      <option name="impArtTech" value="false"/>
      <option name="players" value="ExiledAllies"/>
    </attachment>
```

Setting a breakpoint on `getStrength` to double check when it is loaded could help definitely answer the question.